### PR TITLE
refactor: update react and chartjs versions and dependencies

### PR DESCRIPTION
**Pull request by anthonychoueiri**
Updates React to v18 and all other dependencies to their latest versions.

### CODE REVIEW

**Review by WajeehZantout — CHANGES_REQUESTED** [review]
Hi @anthonychoueiri , I have some comments:

1- I'm seeing the following warning when running `npm start`. Can you please look into it?
```
WARNING in ./src/chart.js 115:47-60
export 'HorizontalBar' (imported as 'HorizontalBar') was not found in 'react-chartjs-2' (possible exports: Bar, Bubble, Chart, Doughnut, Line, Pie, PolarArea, Radar, Scatter, getDatasetAtEvent, getElementAtEvent, getElementsAtEvent)
 @ ./src/index.js 27:0-31 206:18-26
 ```
 
 2- Update lock file version to `2`. It's currently using an old lockfile. This requires updating `node`/`npm` version
 
 3- Remove React import statements as they are not needed in React +17
 
 4- Update `SmartChart` component to a functional component that uses React hooks

**Review by alexiakl — CHANGES_REQUESTED** [review]
horizontalBar chart type was removed. Horizontal bar charts can be configured using the new [indexAxis](https://www.chartjs.org/docs/latest/charts/bar.html#horizontal-bar-chart) option

Source: https://www.chartjs.org/docs/latest/getting-started/v3-migration.html

**Comment by alexiakl** [comment]
Please also add the following to the docs:
```
To use this component, you'll need a Google API key. First **enable** Google Sheets API and then generate the API Key [here](https://console.cloud.google.com/apis/credentials) and restrict it to HTTP referrers (web sites). 
```

**Comment by alexiakl** [comment]
One more thing, please check this SO thread, as I am getting `Uncaught Error: "arc" is not a registered element.` error:
https://stackoverflow.com/questions/70098392/react-chartjs-2-with-chartjs-3-error-arc-is-not-a-registered-element

**Comment by alexiakl** [comment]
One more thing, please check this SO thread, as I am getting `Uncaught Error: "arc" is not a registered element.` error:
https://stackoverflow.com/questions/70098392/react-chartjs-2-with-chartjs-3-error-arc-is-not-a-registered-element

File: src/utils/handleOptions.js
Comment (WajeehZantout) [inline src/utils/handleOptions.js:36]:
```suggestion
  options.scales.x.ticks.callback = (value) => {
    const label = this.chart.scales.x.getLabelForValue(value);
    return `${label}${xsuffix}`;
```

**Review by WajeehZantout — CHANGES_REQUESTED** [review]
Great work 👏  I just left a tiny comment. Let's get @alexiakl's approval before you merge.

Screen Shot 2022-09-22 at 16 14 21 PM

File: src/utils/handleOptions.js
Comment (anthonychoueiri) [reply src/utils/handleOptions.js:36]: `this` is undefined in arrow functions. I tried setting `const that = this` outside the function but it was undefined as well.

Comment (alexiakl) [reply src/utils/handleOptions.js:36]: what about skipping `this`? and writing directly `chart.scales.x.getLabelForValue(value);`? would that work?

Comment (alexiakl) [reply src/utils/handleOptions.js:36]: ah that's a callback function, keep it a function that's fine

Comment (anthonychoueiri) [reply src/utils/handleOptions.js:36]: that wouldn't work. `chart` is undefined.

**Review by WajeehZantout — APPROVED** [review]

**Pull request merged**
